Controllers/HomeController.cs

```diff
@@ -8,100 +9,124 @@
             {
-                UnfinishedTour = _repository.Tours.First(t => t.TourId == tourId);
                 //sets tour to UnfinishedTour variable so we can use it in UserForm 
                 //and then save the finished product to the database
-
-                return RedirectToAction("UserForm", UnfinishedTour.TourId);//if valid returns UserForm
+                var signUp = new SignUpViewModel
+                {
+                    TourId = _repository.Tours
+                           .Where(t => t.TourId == tourId)
+                           .FirstOrDefault().TourId,
+                };
+                return View("UserForm", signUp);//if valid returns UserForm
             }
-            return View();//when submitted
+            
+            // Pass back the same Tour instance if the model is not valid.
+            return View(_repository.Tours.
+                        Where(t => t.TourId == tourId));//when submitted
         }
-        [HttpPost]
-        public IActionResult UserForm(TourParty tourParty)
+
+
+        [HttpGet]
+        public IActionResult UserForm(int tourId)
         {
-            if (ModelState.IsValid)//validates tour
+            _logger.LogInformation("GET request on UserForm.");
+            var signUp = new SignUpViewModel
             {
-                
-                //UnfinishedTour.AddTourist(tourParty);
-                //_repository.Tours.Where(t => t.TourId == UnfinishedTour.TourId);
+                TourId = tourId
+            };
 
-                //Product prodToUpdate = context.Products
-                //     .Where(p => p.ProductID == product.ProductID).FirstOrDefault();
+            // Return the ViewModel with the proper Tour object and a 
+            // new Party to be filled out.
+            return View(signUp);
+        }
 
-                //if (UnfinishedTour != null)
-                
-                Tour tour = _context.Tours.First(t => t.TourId == UnfinishedTour.TourId);
-                tour.AddTourist(tourParty);
+        [HttpPost]
+        public IActionResult UserForm(SignUpViewModel signUp)
+        {
+            _logger.LogInformation("POST request on UserForm.");
+            if (ModelState.IsValid)//validates tour
+            {
+                // If valid, add the Party to the DB.
+                Tour addTour = _repository.Tours
+                               .Where(t => t.TourId == signUp.TourId)
+                               .FirstOrDefault();
+                var party = new Party
+                {
+                    EmailAddress = signUp.EmailAddress,
+                    PartyName = signUp.PartyName,
+                    PhoneNumber = signUp.PhoneNumber,
+                    PartySize = signUp.PartySize,
+                    Tour = addTour,
+                    TourId = signUp.TourId
+                };
+                _context.Parties.Add(party);
```

Comment:
The main issue with the Party not saving was that it was not getting saved to the _context DbSet. Instead it was getting saved to the essentially read only _repository.